src/module/system/action-macros/crafting/repair.ts

```diff
@@ -11,6 +11,12 @@ async function repair(options: RepairActionOptions) {
     const item =
         options.item ?? (options.uuid ? await fromUuid(options.uuid) : await SelectItemDialog.getItem("repair"));
 
+    if (item === null) {
+        // Don't do anything if passed a null item.
+        // TODO: do something smart here. Like possibly rolling the result and the recovered amount both.
+        return;
+    }
+
```

Comment:
This would be handled better in the check right below it.

Reply:
I tried that but it it had two issues.

1) You then get an error from where `item.name` is called because item == null. This can be worked around but makes the following block a bit more messy.
2) This is then returning an error everytime "cancel" is called rather than just exiting out. Which seems like a mistake to me.